### PR TITLE
Add `grafana_state_reason` section in State of alerts

### LINKED ISSUE #82578: Describe in docs what is grafana_state_reason: NoData
### Discussed in https://github.com/grafana/grafana/discussions/68810

<div type='discussions-op-text'>

<sup>Originally posted by **lex-pec** May 22, 2023</sup>
I have some alerts, which by default do not have any data, so their panels show "No Data" in the middle of the graph. 
These alerts are triggered when something appears on the panel - dots of data, usually. So when something appears the alert is triggered and I have a message in my Telegram contact point. When time passes and the time range of the alert leaving that data behind, I am receiving another message, something like: 
``` 
<Alert's name>
grafana_state_reason: NoData

NoData: 
```
Because by default I have no data in my panel a `Nodata = OK` setting is turned on. So I don't understand why I am receiving the latter message.
I've noticed that such a behavior appeared after upgrading Grafana from 9.0.1 to 9.3.6.

Trying to find similar issues or discussions I came across an isssue #57115 which solved a problem of misunderstanding of `grafana_state_reason: MissingSeries`. 

Generally, a similar issue is needed here: to describe how grafana_state_reason: NoData works and the most important is it possible to turn this off and stop receiving this kind of messages when the `NoData = OK` setting is on.

Link to the post on Grafana Community:
https://community.grafana.com/t/alert-returns-grafana-state-reason-nodata-with-nodata-ok-setting/83490
</div>

### PR DESCRIPTION
**Pull request by ppcano**
Document `grafana_state_reason` annotation in [State and health of alerts](https://grafana.com/docs/grafana/latest/alerting/fundamentals/alert-rule-evaluation/state-and-health/)

Fixes https://github.com/grafana/grafana/issues/82578

## Preview

Screenshot 2024-08-06 at 08 56 31

### CODE REVIEW

**Comment by ppcano** [comment]
@yuri-tceretian, the new `grafana_state_reason` section provides an overview of this annotation and a few common examples without detailing all the cases.  Please, let us know if it is correct or if something else should be added. Thanks!

**Review by brendamuir — APPROVED** [review]

File: docs/sources/alerting/fundamentals/alert-rule-evaluation/state-and-health.md [hunk 1]
```diff
@@ -64,18 +64,35 @@ Alert instances will be routed for [notifications](ref:notifications) when they
...
+- If "no data" handling is configured to transition to a state other than `NoData`.
+- If "error" handling is configured to transition to a state other than `Error`.
+
+In these situations, the evaluation state may differ from the alert state, and it might be necessary to understand the reason for being in that state when receiving the notification.
+
+The `grafana_state_reason` annotation is included in these situations, providing the reason in the notifications that explain why the alert instance transitioned to its current state. For example:
+
+- Stale alert instances in the `Normal` state include the `grafana_state_reason` annotation with the value **MissingSeries**.
+- If "no data" or "error" handling transitions to the `Normal` state, the `grafana_state_reason` annotation is included with the value **NoData** or **Error**, respectively.
+
```
[the remark this answers is not in the document]
Comment (ppcano) [reply docs/sources/alerting/fundamentals/alert-rule-evaluation/state-and-health.md:97]: @yuri-tceretian, thanks for pointing out these cases. 

https://github.com/grafana/grafana/pull/91562/commits/b27a4779b34d2a42078268af89192fae0c162368 addresses this. Please, could you review the PR again to get a 👍  to merge this?

**Review by brendamuir — APPROVED** [review]

**Review by yuri-tceretian — APPROVED** [review]
LGTM

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. This is a friendly reminder that the Steady [11.2.x Release Branch](https://github.com/grafana/grafana/tree/v11.2.x) was cut this week, and that it includes the changes committed up to https://github.com/grafana/grafana/commit/b4126d3bce3627e9460e2db0e870c8d984be8e84. This PR was merged after it, so we wanted to let you know that you can add the `backport v11.2.x` label to to backport it and make it part of the next 11.2.x Patch Release, just in case you may have missed the news.

Thank you!

**Pull request merged**